pkg/tsdb/v3iotsdb.go

```diff
@@ -182,34 +182,44 @@ func (a *V3ioAdapter) Querier(_ context.Context, mint, maxt int64) (*querier.V3i
 func (a *V3ioAdapter) DeleteDB(configExists bool, force bool, fromTime int64, toTime int64) error {
 	partitions := a.partitionMngr.PartsForRange(fromTime, toTime)
 	for _, part := range partitions {
-		if toTime == 0 || part.GetEndTime() < toTime {
-			a.logger.Info("Delete partition %s", part.GetTablePath())
-			err := utils.DeleteTable(a.logger, a.container, part.GetTablePath(), "", a.cfg.QryWorkers)
-			if err != nil && !force {
-				return err
-			}
-			// delete the Directory object
-			a.container.Sync.DeleteObject(&v3io.DeleteObjectInput{Path: part.GetTablePath()})
+		a.logger.Info("Delete partition %s", part.GetTablePath())
+		err := utils.DeleteTable(a.logger, a.container, part.GetTablePath(), "", a.cfg.QryWorkers)
+		if err != nil && !force {
+			return errors.Wrap(err, "Failed to delete partition %s"+part.GetTablePath())
+		}
+		// delete the Directory object
+		err = a.container.Sync.DeleteObject(&v3io.DeleteObjectInput{Path: part.GetTablePath()})
+		if err != nil && !force {
+			return errors.Wrap(err, "Failed to delete partition object %s"+part.GetTablePath())
 		}
 	}
-	path := a.cfg.Path + "/names/"
-	a.logger.Info("Delete metric names in path %s", path)
-	err := utils.DeleteTable(a.logger, a.container, path, "", a.cfg.QryWorkers)
-	if err != nil && !force {
-		return err
-	}
-	// delete the Directory object
-	a.container.Sync.DeleteObject(&v3io.DeleteObjectInput{Path: path})
+	a.partitionMngr.DeletePartitionsFromSchema(partitions)
 
+	if len(a.partitionMngr.GetPartitionsPaths()) == 0 {
+		path := a.cfg.Path + "/names/"
+		a.logger.Info("Delete metric names in path %s", path)
+		err := utils.DeleteTable(a.logger, a.container, path, "", a.cfg.QryWorkers)
+		if err != nil && !force {
+			return errors.Wrap(err, "Failed to delete names table")
+		}
+		// delete the Directory object
+		err = a.container.Sync.DeleteObject(&v3io.DeleteObjectInput{Path: path})
+		if err != nil && !force {
+			return errors.Wrap(err, "Failed to delete names object")
```

Comment:
C&P

Reply:
no this is correct, first delete is for the table, second delete is for the object. both of them are for "names"